sky/serve/replica_managers.py

```diff
@@ -582,18 +582,22 @@ def __init__(self, service_name: str, spec: 'service_spec.SkyServiceSpec',
 
     def get_ready_replica_urls(self) -> List[str]:
         ready_replica_urls = []
-        version2url = collections.defaultdict(list)
+        version2url: DefaultDict[int, List[str]] = collections.defaultdict(list)
+        version2count: DefaultDict[int, int] = collections.defaultdict(int)
         for info in serve_state.get_replica_infos(self._service_name):
             if info.status == serve_state.ReplicaStatus.READY:
                 assert info.url is not None
                 version2url[info.version].append(info.url)
+                version2count[info.version] += 1
                 ready_replica_urls.append(info.url)
         # Try all version in descending order. There is possibility that
         # user consecutively update the service several times, and some
         # version might not have any ready replicas.
         version = self.latest_version
         while version >= serve_constants.INITIAL_VERSION:
-            if version in version2url:
+            spec = serve_state.get_spec(self._service_name, version)
+            if spec is not None and version in version2url and version2count[
+                    version] > spec.min_replicas:
```

Comment:
What if there is no version that has `min_replicas`? Should we then return the version with the max ready replicas?

Also, if we only return urls with at lest min replicas, should we scale down those versions smaller than this specific version (`active_version`)?

Reply:
Also, if we only return urls with at lest min replicas, should we scale down those versions smaller than this specific version (active_version)?
=> This should be taken care of by the autoscaler.

Reply:
What if there is no version that has min_replicas? Should we then return the version with the max ready replicas?
=> I am not too sure. (1) return version with max ready replica (2) return URLs with the highest version (3) return no URLs (service temporarily not ready). Which one do you think is better?

Reply:
> Also, if we only return urls with at lest min replicas, should we scale down those versions smaller than this specific version (active_version)? => This should be taken care of by the autoscaler.

Yes, so it seems we don't have such code for autoscaler in this PR. Should we add it?
Another proposal would be letting the autoscaler take care of the `active_version`, so it can know which versions to scale down, and `replica_manager` could just return a dict of version to replicas. The controller could combine the two information and return the correct list of replicas to the load balancer?

> I am not too sure. (1) return version with max ready replica (2) return URLs with the highest version (3) return no URLs (service temporarily not ready). Which one do you think is better?

Returning the latest version with at least one ready replica sounds reasonable to me?

Reply:
Make sense. Updated. Let's do it in the autoscaler (scale down old replicas than `latest_version_with_min_replicas`)

Reply:
To discuss, should we use `latest_spec.min_replicas` instead? Feeling like this could best represent user's requirements at the moment

Reply:
cc @Michaelvll 
I think this is fine since say version 1 launches 2 A100s, and version 2 launches 2 V100s. The `min_replicas` will depend on the resources of each version.

Reply:
I think it might be better to use specific version's `min_replicas`.  It seems that we are using different `min_replicas` in autoscaler vs the replica_manager?